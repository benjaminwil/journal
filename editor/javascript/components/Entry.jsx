import React, {Component} from 'react';

import EntryControls from './EntryControls';
import EntryDateTime from './EntryDateTime';
import EntryTitle from './EntryTitle';

class Entry extends React.Component {
  constructor(props) {
    super(props);
    this.entry = React.createRef();
    this.state = {active: null, displayed: true};
    this.disappearEntry = this.disappearEntry.bind(this);
    this.forceCloseInactiveEntry = this.forceCloseInactiveEntry.bind(this);
  }

  deleteEntry() {
    const url = this.props.urls.show;
    return fetch(url, {
      method: 'DELETE',
    });
  }

  disappearEntry() {
    this.deleteEntry();
    this.setState({displayed: !this.state.displayed});
  }

  forceCloseInactiveEntry() {
    if (this.entry.current.contains(document.activeElement)) {
    } else this.props.deactivateEntry;
  }

  render() {
    return (
      this.state.displayed && (
        <article
          className="entry"
          onBlur={this.forceCloseInactiveEntry}
          onClick={this.props.toggleActivatedEntry}
          onKeyPress={this.props.toggleActivatedEntry}
          ref={this.entry}
          tabIndex="0">
          <EntryTitle title={this.props.title} />
          <EntryDateTime dateTime={this.props.dateTime} />
          <EntryControls
            id={this.props.id}
            active={this.props.active}
            ondelete={this.disappearEntry}
            urls={this.props.urls}
          />
        </article>
      )
    );
  }
}

export default Entry;
