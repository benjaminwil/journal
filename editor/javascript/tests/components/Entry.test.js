import React from 'react';
import renderer from 'react-test-renderer';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

import Entry from '../../components/Entry.jsx';

const props = {
  id: '1',
  title: 'I walked for a while',
  datetime: {created: '2019-09-02T19:47:41.113Z'},
  urls: {edit: '/entries/1/edit', show: '/entries/1'},
};

const entryWrapper = shallow(
  <Entry
    key={props.id}
    title={props.title}
    datetime={props.datetime.created}
    urls={props.urls}
  />,
);

test('the Entry component displays one entry', () => {
  const component = renderer
    .create(
      <Entry
        key={props.id}
        title={props.title}
        datetime={props.datetime.created}
        urls={props.urls}
      />,
    )
    .toJSON();

  expect(component).toMatchSnapshot();
});

test('when first rendered, an Entry does not display EntryControls', () => {
  expect(entryWrapper.find('.controls')).not;
});

test('when clicked, an Entry displays its EntryControls', () => {
  entryWrapper.simulate('click');
  expect(entryWrapper.find('.controls'));
});

test('when clicked and then clicked again, the Entry does not display its \
  EntryControls', () => {
  entryWrapper.simulate('click');
  entryWrapper.simulate('click');
  expect(entryWrapper.find('.controls')).not;
});
