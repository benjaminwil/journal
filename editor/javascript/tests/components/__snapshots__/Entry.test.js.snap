// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Entry component displays one entry 1`] = `
<article
  className="entry"
>
  <h1
    className="title"
  >
    I walked for a while
  </h1>
  <p />
</article>
`;
